patch-Always-allocate-domid-sequentially-and-do-not-reuse-.patch

```diff
@@ -0,0 +1,57 @@
+From fa4ac191b98bdfa163f4b49b43536aad0b098518 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
+ <marmarek@invisiblethingslab.com>
+Date: Fri, 3 Dec 2021 21:13:23 +0100
+Subject: [PATCH] Always allocate domid sequentially, and do not reuse them
+
+While domid wrapping around is unlikely on Qubes, make sure it wont
+happen, to avoid various corner cases.
+---
+ xen/common/domctl.c | 29 ++++++++---------------------
+ 1 file changed, 8 insertions(+), 21 deletions(-)
+
+diff --git a/xen/common/domctl.c b/xen/common/domctl.c
+index 2b25653b49..ed63abd312 100644
+--- a/xen/common/domctl.c
++++ b/xen/common/domctl.c
+@@ -491,29 +491,16 @@ long do_domctl(XEN_GUEST_HANDLE_PARAM(xen_domctl_t) u_domctl)
+         domid_t        dom;
+         static domid_t rover = 0;
+ 
+-        dom = op->domain;
+-        if ( (dom > 0) && (dom < DOMID_FIRST_RESERVED) )
+-        {
+-            ret = -EEXIST;
+-            if ( !is_free_domid(dom) )
+-                break;
++        if ( rover >= DOMID_FIRST_RESERVED ) {
++            printk(XENLOG_ERR
++                   "domctl: out of available domid values, reboot the system\n");
++            return -ENOMEM;
+         }
+-        else
+-        {
+-            for ( dom = rover + 1; dom != rover; dom++ )
+-            {
+-                if ( dom == DOMID_FIRST_RESERVED )
+-                    dom = 1;
+-                if ( is_free_domid(dom) )
+-                    break;
+-            }
+-
+-            ret = -ENOMEM;
+-            if ( dom == rover )
+-                break;
+ 
+-            rover = dom;
+-        }
++        /* Intentionally ignore op->domain */
```

Comment:
Is there a case for which we need this? Otherwise I think it would be better to return an error. IIUC the previous behavior guaranteed that if you requested a specific domid and createdomain returned no error the new domain got that domid.